File: autocard/frontend/src/cad/contracts/layers.ts
export type LineType =
  | 'continuous'
  | 'dashed'
  | 'dotted'
  | 'dashdot'
  | 'center'
  | 'phantom'
  | 'hidden'

export type LayerDef = {
  id: string
  name: string
  visible: boolean
  locked: boolean
  frozen: boolean
  color: string
  lineType: LineType
  lineWeight: number
  plotStyle?: string
  parentId?: string
}

export type StyleDef = {
  id: string
  name: string
  color?: string
  lineType?: LineType
  lineWeight?: number
  fillColor?: string
  fontSize?: number
  fontFamily?: string
  textAlign?: 'left' | 'center' | 'right'
  arrowStyle?: 'open' | 'closed' | 'dot' | 'tick'
}

export const DEFAULT_AEC_LAYERS: Omit<LayerDef, 'id'>[] = [
  { name: 'A-WALL',  visible: true, locked: false, frozen: false, color: '#FFFFFF', lineType: 'continuous', lineWeight: 0.5 },
  { name: 'A-DOOR',  visible: true, locked: false, frozen: false, color: '#4DA6FF', lineType: 'continuous', lineWeight: 0.35 },
  { name: 'A-WIND',  visible: true, locked: false, frozen: false, color: '#4DA6FF', lineType: 'continuous', lineWeight: 0.25 },
  { name: 'A-ROOM',  visible: true, locked: false, frozen: false, color: '#FFD700', lineType: 'continuous', lineWeight: 0.18 },
  { name: 'A-DIMS',  visible: true, locked: false, frozen: false, color: '#FF8C00', lineType: 'continuous', lineWeight: 0.18 },
  { name: 'A-ANNO',  visible: true, locked: false, frozen: false, color: '#CCCCCC', lineType: 'continuous', lineWeight: 0.18 },
  { name: 'A-GRID',  visible: true, locked: false, frozen: false, color: '#555555', lineType: 'dashdot',    lineWeight: 0.18 },
  { name: 'A-HATCH', visible: true, locked: false, frozen: false, color: '#888888', lineType: 'continuous', lineWeight: 0.13 },
]

File: autocard/backend/cad/schema/layers.go
// mirrors cad/contracts/layers.ts
package schema

type LayerDef struct {
	ID         string  `json:"id"`
	Name       string  `json:"name"`
	Visible    bool    `json:"visible"`
	Locked     bool    `json:"locked"`
	Frozen     bool    `json:"frozen"`
	Color      string  `json:"color"`
	LineType   string  `json:"lineType"`
	LineWeight float64 `json:"lineWeight"`
	PlotStyle  string  `json:"plotStyle,omitempty"`
	ParentID   string  `json:"parentId,omitempty"`
}

type StyleDef struct {
	ID          string  `json:"id"`
	Name        string  `json:"name"`
	Color       string  `json:"color,omitempty"`
	LineType    string  `json:"lineType,omitempty"`
	LineWeight  float64 `json:"lineWeight,omitempty"`
	FillColor   string  `json:"fillColor,omitempty"`
	FontSize    float64 `json:"fontSize,omitempty"`
	FontFamily  string  `json:"fontFamily,omitempty"`
	TextAlign   string  `json:"textAlign,omitempty"`
	ArrowStyle  string  `json:"arrowStyle,omitempty"`
}

type SheetDef struct {
	ID               string   `json:"id"`
	Name             string   `json:"name"`
	Width            float64  `json:"width"`
	Height           float64  `json:"height"`
	Scale            float64  `json:"scale"`
	ViewportIDs      []string `json:"viewportIds"`
	TitleBlockNodeID string   `json:"titleBlockNodeId,omitempty"`
	DrawingNumber    string   `json:"drawingNumber,omitempty"`
	Revision         string   `json:"revision,omitempty"`
}

type ViewDef struct {
	ID             string                       `json:"id"`
	Name           string                       `json:"name"`
	Center         Point                        `json:"center"`
	Zoom           float64                      `json:"zoom"`
	LayerOverrides map[string]map[string]interface{} `json:"layerOverrides,omitempty"`
}

type ConstraintDef struct {
	ID         string                 `json:"id"`
	Type       string                 `json:"type"`
	NodeIDs    []string               `json:"nodeIds"`
	Params     map[string]interface{} `json:"params,omitempty"`
	Satisfied  bool                   `json:"satisfied"`
	Persistent bool                   `json:"persistent"`
}
